Blish HUD/GameServices/Modules/Pkgs/GitHubPkgRepoProvider.cs

```diff
@@ -0,0 +1,153 @@
+            var manifests = await LoadPkgManifests(releaseResult.Releases);
+
+            if (manifests.Exception != null) {
+                progress?.Report($"{Strings.GameServices.ModulesService.PkgManagement_Progress_FailedToReadOrParseRepoManifest}\r\n{manifests.Exception.Message}");
+                return null;    
+            }
+
+            return manifests.PkgManifests;
+        }
+
+        private async Task<(GitHubRelease[] Releases, Exception Exception)> UpdateGitHubReleases() {
+            try {
+                return (await $"{this.RepoUrl}{GITHUB_RELEASES_URI}".WithHeader("User-Agent", "Blish-HUD").GetJsonAsync<GitHubRelease[]>(), null);
+            } catch (FlurlHttpException ex) {
+                return (new GitHubRelease[0], ex);
+            }
+        }
+
+        private async Task<(PkgManifest[] PkgManifests, Exception Exception)> LoadPkgManifests(IEnumerable<GitHubRelease> releases) {
+            Exception lastException = null;
+
+            foreach (var release in releases) {
+                string compressedReleaseUrl = release.Assets.First(asset => asset.Name.Equals(ASSET_PACKAGE_NAME, StringComparison.InvariantCultureIgnoreCase)).BrowserDownloadUrl;
+
+                try {
+                    using var compressedRelease = await compressedReleaseUrl.GetStreamAsync();
+
+                    using var gzipStream = new GZipStream(compressedRelease, CompressionMode.Decompress);
+                    using var streamReader = new StreamReader(gzipStream);
+                    using var jsonTextReader = new JsonTextReader(streamReader);
+                    var serializer = new JsonSerializer();
+
+                    return (serializer.Deserialize<PkgManifest[]>(jsonTextReader), null);
+                } catch (Exception ex) {
+                    Logger.Warn(ex, $"Failed to load release list from '{compressedReleaseUrl}'.");
+                    lastException = ex;
+                }
+            }
+
+            return (null, lastException);
+        }
+
+        public IEnumerable<PkgManifest> GetPkgManifests() {
+            return _pkgCache[this.RepoUrl].Where(pkg => _activeFilters.All(filter => filter(pkg)));
+        }
+
+        public IEnumerable<(string OptionName, Action<bool> OptionAction, bool IsToggle)> GetExtraOptions() {
+            yield return ("Force Reload Repository", async (toggleState) => { await LoadRepo(); }, false);
+
+            yield return ("Only Show Supported Versions", (toggleState) => ToggleFilter(FilterShowOnlySupportedVersion, toggleState), true);
+            yield return ("Only Show Modules With Updates", (toggleState) => ToggleFilter(FilterShowOnlyUpdates,        toggleState), true);
+            yield return ("Only Show Installed Modules", (toggleState) => ToggleFilter(FilterShowOnlyInstalled,         toggleState), true);
+            yield return ("Only Show Modules Not Installed", (toggleState) => ToggleFilter(FilterShowOnlyNotInstalled,  toggleState), true);
```

Comment:
Not a blocker, but in the future this should support localization

Reply:
e1ee816 Thanks!